ibis/backends/base/__init__.py

```diff
@@ -593,3 +595,28 @@ def has_operation(cls, operation: type[ops.Value]) -> bool:
         raise NotImplementedError(
             f"{cls.name} backend has not implemented `has_operation` API"
         )
+
+
+def connect(url_or_path: str) -> BaseBackend:
+    'Connect to given *url_or_path*. Auto-detect backend from url scheme or path extension.'
+    path = Path(url_or_path)
+
+    if '://' in url_or_path:
+        r = urllib.parse.urlparse(url_or_path)
+        path = Path(r.path)
+
+        schemeparts = r.scheme.split('+')
+        if 'ibis' in schemeparts:
+            schemeparts.remove('ibis')
+        if '' in schemeparts:
+            schemeparts.remove('')
+
+        if schemeparts:
+            return getattr(ibis, schemeparts[0]).connect(r.path)
+
+    ext = path.suffix[1:]
+
+    if ext in {'parquet', 'csv'}:
+        return ibis.datafusion.connect({path.name: path})
```

Comment:
We probably want to use DuckDB for both of these since Datafusion operation support is pretty sparse right now. Something like this:

```python
con = ibis.duckdb.connect()  # in memory connection
con.register_csv(...)  # this API doesn't exist yet
# or 
con.register_parquet(...)  # also doesn't exist yet
```